mlflow/cli.py

```diff
@@ -144,6 +144,20 @@ def run(uri, entry_point, version, param_list, experiment_name, experiment_id, m
         _logger.error("=== %s ===", e)
         sys.exit(1)
 
+def _validate_server_args(static_prefix=None, gunicorn_opts=None, workers=None, waitress_opts=None):
+    server_kwargs = {'static_prefix': static_prefix}
+    if sys.platform == "win32":
+        if gunicorn_opts is not None:
+            raise NotImplementedError("waitress replaces gunicorn on Windows, cannot specify --gunicorn-opts")
+        server_kwargs.update({
+            'waitress_opts': waitress_opts
+        })
+    else:
+        server_kwargs.update({
```

Comment:
should we raise exception if we got waitress options similarly to above?

Reply:
Yes, that's a great idea

Reply:
Switch to a pure validation function, but tweaked the default handling for "workers" to detect a manual selection of the default value - please take a look and let me know what you think!